crates/oxc_semantic/src/symbol/table.rs

```diff
@@ -1,16 +1,16 @@
 use std::ops::{Deref, Index, IndexMut};
 
 use oxc_ast::{Atom, Span};
-use rustc_hash::FxHashMap;
 
+use super::reference::ResolvedReferenceId;
 use super::{Symbol, SymbolFlags, SymbolId};
 use crate::node::AstNodeId;
-use crate::ResolvedReference;
+use crate::{Reference, ResolvedReference};
 
 #[derive(Debug, Default)]
 pub struct SymbolTable {
     symbols: Vec<Symbol>,
-    resolved_references: FxHashMap<AstNodeId, ResolvedReference>,
+    resolved_references: Vec<ResolvedReference>,
```

Comment:
We need to document all these fields at some point, the symbol table is starting to look like a database with indexes everywhere 🤣

Reply:
We should add a comment here indicating this is an index, the index is its id. We can wrap it in a newtype later.

Reply:
I added docs to make it clear.